src/main/java/com/thevoxelbox/voxelsniper/brush/GenerateTreeBrush.java

```diff
@@ -293,47 +250,40 @@ private void trunkCreate() {
 
             for (int z = this.thickness; z >= 0; z--) {
                 if ((xSquared + Math.pow(z, 2)) <= bSquared) {
-                    // If block is air, then create a block.
-                    if (this.getWorld().getBlockAt(blockPositionX + x, blockPositionY, blockPositionZ + z).getType() == Material.AIR) {
-                        // Adds block to undo function.
-                        if (this.getBlockMaterialAt(blockPositionX + x, blockPositionY, blockPositionZ + z) != woodMaterial) {
-                            this.undo.put(this.clampY(blockPositionX + x, blockPositionY, blockPositionZ + z));
-                        }
-                        // Creates block.
-                        this.clampY(blockPositionX + x, blockPositionY, blockPositionZ + z).setBlockData(woodMaterial.createBlockData(), false);
-                    }
-                    if (this.getWorld().getBlockAt(blockPositionX + x, blockPositionY, blockPositionZ - z).getType() == Material.AIR) {
-                        if (this.getBlockMaterialAt(blockPositionX + x, blockPositionY, blockPositionZ - z) != woodMaterial) {
-                            this.undo.put(this.clampY(blockPositionX + x, blockPositionY, blockPositionZ - z));
-                        }
-                        this.clampY(blockPositionX + x, blockPositionY, blockPositionZ - z).setBlockData(woodMaterial.createBlockData(), false);
-                    }
-                    if (this.getWorld().getBlockAt(blockPositionX - x, blockPositionY, blockPositionZ + z).getType() == Material.AIR) {
-                        if (this.getBlockMaterialAt(blockPositionX - x, blockPositionY, blockPositionZ + z) != woodMaterial) {
-                            this.undo.put(this.clampY(blockPositionX - x, blockPositionY, blockPositionZ + z));
-                        }
-                        this.clampY(blockPositionX - x, blockPositionY, blockPositionZ + z).setBlockData(woodMaterial.createBlockData(), false);
-                    }
-                    if (this.getWorld().getBlockAt(blockPositionX - x, blockPositionY, blockPositionZ - z).getType() == Material.AIR) {
-                        if (this.getBlockMaterialAt(blockPositionX - x, blockPositionY, blockPositionZ - z) != woodMaterial) {
-                            this.undo.put(this.clampY(blockPositionX - x, blockPositionY, blockPositionZ - z));
+                    for (int i = 0; i < 4; i++) {
+                        switch (i) {
```

Comment:
Same story again
https://github.com/KevinDaGame/VoxelSniper/blob/master/src/main/java/com/thevoxelbox/voxelsniper/brush/DrainBrush.java#L43-L45

Reply:
Same story as what?

Reply:
This is already a lot better then it was, but could we perhaps do it with three for loops instead? (like how I did it [here](https://github.com/KevinDaGame/VoxelSniper/blob/master/src/main/java/com/thevoxelbox/voxelsniper/brush/DrainBrush.java#L43-L45)).

Reply:
Fixed